hermes-consumers/src/test/java/pl/allegro/tech/hermes/consumers/consumer/batch/DirectBufferPoolTest.java

```diff
@@ -0,0 +1,215 @@
+package pl.allegro.tech.hermes.consumers.consumer.batch;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.nio.BufferOverflowException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.*;
+
+public class DirectBufferPoolTest {
```

Comment:
I don't have good answer for this. I can rewrite that spock test in junit for the sake of consistency if you want...